### PR TITLE
REST interface exposed by W

### PR DESCRIPTION
**Pull request by malishav**
The PR removes Section 5 - ACE profile. Instead, it introduces an initial draft of the REST interface exposed by W. Comments are welcome on what else needs to be specified here.

### CODE REVIEW

**Comment by gselander** [comment]
Just had a first glance. One question: 

Why is "TODO: register rsp_ad ACE parameter" still in?

File: draft-selander-ace-ake-authz.md [hunk 1]
```diff
@@ -476,142 +478,37 @@ V receives the voucher response from W over the secure connection. If the receiv
...
+The interaction between V and W is enabled through a RESTful HTTPS interface exposed by W.
+V MUST perform a TLS handshake with W, reachable at the URI specified in LOC_W.
+V MUST use the TLS client authentication to authenticate to W, using the certificate containing the PK_V public key.
 
-* U plays the role of the ACE Resource Server (RS).
+## HTTP URIs
 
-* V plays the role of the ACE Client (C).
+W MUST support the use of the path-prefix "/.well-known/", as defined in {{RFC5785}}, and the registered name "ake-authz".
+A valid URI thus begins with "https://www.example.com/.well-known/ake-authz".
```
Comment (mcr) [inline draft-selander-ace-ake-authz.md:490]: Could go under /.well-known/brski

File: draft-selander-ace-ake-authz.md [hunk 2]
```diff
@@ -476,142 +478,37 @@ V receives the voucher response from W over the secure connection. If the receiv
...
+To request a voucher, V MUST issue an HTTP request:
 
-C and RS use EDHOC's EAD to communicate.
-C and RS use the EDHOC protocol to protect their communication.
-EDHOC also provides mutual authentication of C and RS, assisted by the AS.
+* Method is POST
+* Payload is the base64-encoded serialization of the Voucher Request object, as specified in {{voucher_request}}.
 
-## Protocol Overview
+In case of successful processing at W, W MUST issue a 200 OK response with payload containing base64-encoded Voucher Response object, as specified in {{voucher_response}}.
```
Comment (mcr) [inline draft-selander-ace-ake-authz.md:501]: why is it base64 encoded?

Comment (malishav) [reply draft-selander-ace-ake-authz.md:501]: I thought it was necessary but it seems I am wrong. Does the rest of the PR look good? Anything else that needs to be specified?

Comment (mcr) [reply draft-selander-ace-ake-authz.md:501]: I am not sure if there is anything else, but merge it.

**Review by mcr — APPROVED** [review]

File: draft-selander-ace-ake-authz.md (hunk 2 above)
Comment (malishav) [reply draft-selander-ace-ake-authz.md]: fixed

File: draft-selander-ace-ake-authz.md (hunk 1 above)
Comment (gselander) [reply draft-selander-ace-ake-authz.md:489]: ... or /.well-known/edhoc

I think we can keep ake-authz for the time being.

File: draft-selander-ace-ake-authz.md [hunk 3]
```diff
@@ -476,142 +477,36 @@ V receives the voucher response from W over the secure connection. If the receiv
 
 V verifies CRED_I and that U is an admissible device and then continues the EDHOC processing, or else discontinues the protocol.
 
-# ACE Profile
+# REST Interface at W
 
-The messages specified in this document may be carried between the endpoints in various protocols. This section defines an embedding as a profile of the ACE framework (see Appendix C of {{I-D.ietf-ace-oauth-authz}}).
+The interaction between V and W is enabled through a RESTful HTTPS interface exposed by W.
+V MUST perform a TLS handshake with W, reachable at the URI specified in LOC_W.
+V MUST use the TLS client authentication to authenticate to W, using the certificate containing the PK_V public key.
```
Comment (gselander) [inline draft-selander-ace-ake-authz.md:484]: I think we should be open for other ways to protect REST. Can we change to SHOULD and also mention secured CoAP?

**Review by gselander — APPROVED** [review]

**Comment by gselander** [comment]
Wrong branch! Sorry, my last commit 111cc493 should have been on the master. Too many drafts and branches in parallel . . .

**Comment by malishav** [comment]
@gselander see if the new commit makes sense, I force pushed your commit by mistake

**Comment by gselander** [comment]
This is fine. I was thinking EDHOC and OSCORE as well, but that is not excluded, we can add that in the master. I'll merge it now.

**Pull request merged**